### ISSUE TITLE
Course and heading

### ISSUE THREAD

**Issue by fvantienen** [Fabien-B/GcsAgain#16]
It would be nice if we could display both the course and heading of the UAV's and the GCS on the map. Maybe we can add a tiny arrow pointing at the course?
Currently we only display heading, but the course can show the crap angle

**Comment by Fabien-B** [comment]
Until now we were relying on the position history, but its true that its not very accurate, in particular when the position update rate is low.
That could probably be added in the `GraphicsAircrafts` paint method.
See https://github.com/paparazzi/PprzGCS/blob/master/src/widgets/map/graphics_objects/graphics_aircraft.cpp#L23-L28
and: https://github.com/paparazzi/PprzGCS/blob/master/src/widgets/map/graphics_objects/graphics_intruder.cpp#L24-L27
The AircraftItem is currently rotated by `heading`, so the course arrow should probably be inclined by  `route-heading` to be coherent.
: https://github.com/paparazzi/PprzGCS/blob/master/src/widgets/map/map_items/aircraft_item.cpp#L55

**Comment by fvantienen** [comment]
Quick question how does the ground GPS gets drawn in the GCS? Since for example `ublox2ivy` sends a `FLIGHT_PARAM` message with id `0`, but I can only find `MapWidget::updateAircraftItem` listening for that and ignoring it if there is no aircraft. But if I'm correct it is the symbol of the intruder? So how does this all work?

**Comment by noether** [comment]
Indeed, if the compass option (or any other flag) is activated, it makes sense and is a good idea. Otherwise, I do not think so.

It is in the Python App to display the GVF.

![image](https://github.com/paparazzi/PprzGCS/assets/148152/e0ccc78f-7c13-4b6e-b673-2ba83a338d9e)

**Comment by Fabien-B** [comment]
Hi,
The GCS position is draw in  `MapWidget::onGCSPos`, which is called from a direct binding to the `FLIGHT_PARAM` message.
I am not very happy about the code architecture of the map, I may change it in the future in a V2, but it will take a long time.

**Comment by OpenUAS** [comment]
I appreciate all work done, maybe an idea to close this issue and put it on @Fabien-B Todo item list? since "but it will take a long time." is relative ;)

**Comment by Fabien-B** [comment]
> I am not very happy about the code architecture of the map, I may change it in the future in a V2, but it will take a long time.

The big changes I was referring to are not coming in the near future, so it's completely worthwhile adding this feature if you need it.
You can make it an "option" by adding a checkbox in the map menu and the context menu of the map. The "Show crash prediction" option is a good example to be reproduced.
